test/puppetlabs/puppetdb/middleware_test.clj

```diff
@@ -247,3 +247,42 @@
                        :query-params {"foo" "bar"}}]
           (is (= post-req (middleware-fn post-req)))
           (is (= get-req (middleware-fn get-req))))))))
+
+(deftest verify-sync-version-test
```

Comment:
Have some questions about the tests, but think I'll chat with you about those independently.